KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/QueryParser.java

```diff
@@ -0,0 +1,63 @@
+package org.sleuthkit.autopsy.keywordsearch;
+
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrRequest;
+import org.apache.solr.client.solrj.response.QueryResponse;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Parse query using Solr
+ */
+class QueryParser {
+
+  static class Result {
+    /**
+     * field name -> [term]
+     */
+    Map<String, List<String>> fieldTermsMap;
+  }
+
+  /**
+   * Parse the given query string on Solr and return the result
+   */
+  static Result parse(String query, List<Server.Schema> fields) throws KeywordSearchModuleException, NoOpenCoreException {
+    SolrQuery q = new SolrQuery();
+    q.setShowDebugInfo(true);
```

Comment:
Does debug info really need to be turned on here?

Reply:
Yes. I use the debug flag to get parsed query information here.
https://lucene.apache.org/solr/7_5_0//solr-core/org/apache/solr/handler/FieldAnalysisRequestHandler.html
But there is the non-debug API `FieldAnalysisRequestHandler` for the same purpose. I will come up with a commit which uses it.

Reply:
Changed to use `FieldAnalysisRequestHandler`. The result is the same but it's a official API (not debug).